TrackletAlgorithm/TrackFitMemory.h

```diff
@@ -4,158 +4,909 @@
+  typedef ap_int<TrackFitBase::kTFPhiResidSize> TFSTUB7PHIRESID;
+  typedef ap_int<TrackFitBase::kTFRResidSize> TFSTUB7RRESID;
+
+  typedef ap_uint<TrackFitBase::kTrackWordSize> TrackWord;
+  typedef ap_uint<TrackFitBase::kBarrelStubSize> BarrelStubWord;
+  typedef ap_uint<TrackFitBase::kDiskStubSize> DiskStubWord;
+  typedef ap_uint<TrackFitBase::kTrackFitSize> TrackFitData;
 
   // Constructors
   TrackFit(const TrackFitData& newdata):
     data_(newdata)
   {}
 
-  TrackFit(const TFPT pt, const TFPHI0 phi0, const TFT t, const TFZ0 z0, const TFSTUBINDEX id1, const TFSTUBINDEX id2, const TFSTUBINDEX id3, const TFSTUBINDEX id4, const TFSTUBINDEX id5, const TFSTUBINDEX id6):
-    data_( (((((((((pt,phi0),t),z0),id1),id2),id3),id4),id5),id6) )
+  TrackFit(const TFTRACKINDEX trackIndex, const TFSEEDTYPE seedtype) :
+    data_( ((((((((((((((((((((((((((((((((((((((((((((((((((((((0,seedtype),TFRINV(0)),TFPHI0(0)),TFZ0(0)),TFT(0)),TFLMAP(0)),
+             TFVALID(0)),trackIndex),TFSTUBINDEX(0)),TFSTUB0R(0)),TFSTUB0PHIRESID(0)),TFSTUB0ZRESID(0)),
+             TFVALID(0)),trackIndex),TFSTUBINDEX(0)),TFSTUB1R(0)),TFSTUB1PHIRESID(0)),TFSTUB1ZRESID(0)),
+             TFVALID(0)),trackIndex),TFSTUBINDEX(0)),TFSTUB2R(0)),TFSTUB2PHIRESID(0)),TFSTUB2ZRESID(0)),
+             TFVALID(0)),trackIndex),TFSTUBINDEX(0)),TFSTUB3R(0)),TFSTUB3PHIRESID(0)),TFSTUB3ZRESID(0)),
+             TFVALID(0)),trackIndex),TFSTUBINDEX(0)),TFSTUB4R(0)),TFSTUB4PHIRESID(0)),TFSTUB4RRESID(0)),
+             TFVALID(0)),trackIndex),TFSTUBINDEX(0)),TFSTUB5R(0)),TFSTUB5PHIRESID(0)),TFSTUB5RRESID(0)),
+             TFVALID(0)),trackIndex),TFSTUBINDEX(0)),TFSTUB6R(0)),TFSTUB6PHIRESID(0)),TFSTUB6RRESID(0)),
+             TFVALID(0)),trackIndex),TFSTUBINDEX(0)),TFSTUB7R(0)),TFSTUB7PHIRESID(0)),TFSTUB7RRESID(0)) )
+
   {}
 
-  TrackFit():
+  TrackFit() :
     data_(0)
   {}
 
   #ifndef __SYNTHESIS__
-  TrackFit(const char* datastr, int base=16)
+  TrackFit(const char* datastr, int base = 16)
   {
     TrackFitData newdata(datastr, base);
     data_ = newdata;
   }
   #endif
 
   // Getter
-  static constexpr int getWidth() {return kTrackFitSize;}
+  static constexpr int getWidth() {return TrackFitBase::kTrackFitSize;}
 
   TrackFitData raw() const {return data_;}
 
-  TFPT getPt() const {
-    return data_.range(kTFpTMSB,kTFpTLSB);
+  TFVALID getTrackValid() const {
+    return data_.range(kTFTrackValidMSB,kTFTrackValidLSB);
+  }
+
+  TFSEEDTYPE getSeedType() const {
+    return data_.range(kTFSeedTypeMSB,kTFSeedTypeLSB);
+  }
+
+  TFRINV getRinv() const {
+    return data_.range(kTFRinvMSB,kTFRinvLSB);
   }
 
   TFPHI0 getPhi0() const {
     return data_.range(kTFPhi0MSB,kTFPhi0LSB);
   }
 
+  TFZ0 getZ0() const {
+    return data_.range(kTFZ0MSB,kTFZ0LSB);
+  }
+
   TFT getT() const {
     return data_.range(kTFTMSB,kTFTLSB);
   }
 
-  TFZ0 getZ0() const {
-    return data_.range(kTFZ0MSB,kTFZ0LSB);
+  TFLMAP getLMap() const {
+    return data_.range(kTFLMapMSB,kTFLMapLSB);
```

Comment:
"LMap" is not very descriptive. Can we find better name? e.g. NhitsVsLay. And add comment to function getLMap() explaining what it returns, as not obvious from name.

Reply:
I changed to "HitMap" and added the comment you suggested. Hopefully that name is more meaningful, while still being concise.